app/lib/import_export.rb

```diff
@@ -153,16 +153,13 @@ def update_assessment_symptom_data(data, patients)
     data[:assessments][:checked].delete(:symptoms)
     data[:assessments][:headers].delete('Symptoms Reported')
 
-    # For large queries (e.g. 100k patients) of the `Symptom.where(...) query below
-    # we saw MySQL give up on queries and throw an exception.
-    # The workaround for this is to collect the symptom ids in a set and find them
-    # using batches of patients.
-    symptom_ids = Set.new
-    patients.pluck(:id).in_groups_of(10_000, false) do |batch|
-      symptom_ids << Symptom.where(condition_id: ReportedCondition.where(assessment_id: Assessment.where(patient_id: batch).pluck(:id)).pluck(:id)).pluck(:id)
-    end
-    symptom_names_and_labels = Symptom.where(id: symptom_ids.to_a.flatten).where.not(label: nil).where.not(name: nil)
-                                      .distinct.order(:label).pluck(:name, :label).transpose
+    symptom_names_and_labels = patients.joins(assessments: { reported_condition: :symptoms })
+                                       .where.not(label: nil)
```

Comment:
```
ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'patients.label' in 'where clause')
```

Reply:
```
patients.joins(assessments: { reported_condition: :symptoms }).where('symptoms.name IS NOT NULL AND symptoms.label IS NOT NULL').distinct.order(:label).pluck(:name, :label).transpose
```

Reply:
so I think the problem with this query (where we do a bunch of joining instead of plucking ids) is that the query is noticeably slower (this is something I noticed when working on the export performance ticket)

Would you be able to do benchmark comparisons on the 500k database between:

```
Patient.all.joins(assessments: { reported_conditio
n: :symptoms }).where('symptoms.name IS NOT NULL AND symptoms.label IS NOT NULL').dist
inct.order(:label).pluck(:name, :label).transpose
```

and

```
Symptom.where(condition_id: ReportedCondition.where(assessment_id: Assess
ment.where(patient_id: Patient.all.pluck(:id)).pluck(:id)).pluc
k(:id)).where.not(name: nil).where.not(label: nil).distinct.order(:label).pluck(:name,
 :label).transpose
```

I know the id plucking solution also leads to the problem that @Bialogs solved with his workaround, but maybe we'll have to come up with something else that's clever.